### PR TITLE
llvmPackages_git: update to 19.0.0-unstable-2024-06-12

### PR DESCRIPTION
**Pull request by RossComputerGuy**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by RossComputerGuy** [comment]
Will be updated to the latest later today

**Comment by RossComputerGuy** [comment]
@ofborg build llvmPackages_git.clang

**Comment by Nanotwerp** [comment]
It appears that `libcxx` fails to build.

```
FAILED: libcxxabi/src/CMakeFiles/cxxabi_static_objects.dir/cxa_virtual.cpp.o 
/nix/store/19sfx69d3inys39h88m2b045sasln0dv-clang-wrapper-19.0.0-unstable-2024-05-14/bin/clang++ -DHAVE___CXA_THREAD_ATEXIT_IMPL -DLIBCXX_BUILDING_LIBCXXABI -D_LIBCPP_BUILDING_LIBRARY -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBCXXABI_BUILDING_LIBRARY -D_LIBCXXABI_LINK_PTHREAD_LIB -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/libcxx-src-19.0.0-unstable-2024-05-14/libcxxabi/../libcxx/src -I/build/libcxx-src-19.0.0-unstable-2024-05-14/runtimes/build/include/c++/v1 -I/build/libcxx-src-19.0.0-unstable-2024-05-14/libcxxabi/include -fvisibility-inlines-hidden -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -fdiagnostics-color  -O3 -DNDEBUG -std=c++23 -nostdinc++ -fstrict-aliasing -funwind-tables -D_DEBUG -UNDEBUG -Wall -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error -MD -MT libcxxabi/src/CMakeFiles/cxxabi_static_objects.dir/cxa_virtual.cpp.o -MF libcxxabi/src/CMakeFiles/cxxabi_static_objects.dir/cxa_virtual.cpp.o.d -o libcxxabi/src/CMakeFiles/cxxabi_static_objects.dir/cxa_virtual.cpp.o -c /build/libcxx-src-19.0.0-unstable-2024-05-14/libcxxabi/src/cxa_virtual.cpp
clang++: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
In file included from /build/libcxx-src-19.0.0-unstable-2024-05-14/libcxxabi/src/cxa_virtual.cpp:9:
In file included from /build/libcxx-src-19.0.0-unstable-2024-05-14/runtimes/build/include/c++/v1/cxxabi.h:17:
/build/libcxx-src-19.0.0-unstable-2024-05-14/runtimes/build/include/c++/v1/stddef.h:35:15: fatal error: 'stddef.h' file not found
   35 | #include_next <stddef.h>
      |               ^~~~~~~~~~
1 error generated.
```

**Comment by Nanotwerp** [comment]
> > It appears that `libcxx` fails to build.
> 
> Yeah, I noticed that as well when I got the chance to try building this morning. Not sure why it broke, I'll take a look at it later today when I have time.

I have no idea about the build internals of `LLVM`, but the issue could *possibly* be due to [this commit](https://github.com/llvm/llvm-project/commit/3c4b673af05f53e8a4d1a382b5c86367ea512c9e).

**Comment by RossComputerGuy** [comment]
> > > It appears that `libcxx` fails to build.
> > 
> > 
> > Yeah, I noticed that as well when I got the chance to try building this morning. Not sure why it broke, I'll take a look at it later today when I have time.
> 
> I have no idea about the build internals of `LLVM`, but the issue could _possibly_ be due to [this commit](https://github.com/llvm/llvm-project/commit/3c4b673af05f53e8a4d1a382b5c86367ea512c9e).

I don't think that would cause this issue since that was committed on April 11th. This PR was created on April 14th and built in that revision without this error. It's something more recent.

File: pkgs/development/compilers/llvm/git/clang/gnu-install-dirs.patch [hunk 1]
```diff
@@ -1,5 +1,5 @@
 diff --git a/cmake/modules/AddClang.cmake b/cmake/modules/AddClang.cmake
```
Comment (AndersonTorres) [inline pkgs/development/compilers/llvm/git/clang/gnu-install-dirs.patch:1]: what does this patch do?

Comment (RossComputerGuy) [reply pkgs/development/compilers/llvm/git/clang/gnu-install-dirs.patch:1]: Installs some of clang's components right instead of how they do it upstream.

**Comment by Nanotwerp** [comment]
> > > > It appears that `libcxx` fails to build.
> > > 
> > > 
> > > Yeah, I noticed that as well when I got the chance to try building this morning. Not sure why it broke, I'll take a look at it later today when I have time.
> > 
> > 
> > I have no idea about the build internals of `LLVM`, but the issue could _possibly_ be due to [this commit](https://github.com/llvm/llvm-project/commit/3c4b673af05f53e8a4d1a382b5c86367ea512c9e).
> 
> I don't think that would cause this issue since that was committed on April 11th. This PR was created on April 14th and built in that revision without this error. It's something more recent.

Ah, I see! 

I fear it also fails for `libunwind` too, for the same reason:

```
[1/21] Building CXX object libunwind/src/CMakeFiles/unwind_shared_objects.dir/Unwind-EHABI.cpp.o
clang++: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
[2/21] Building CXX object libunwind/src/CMakeFiles/unwind_static_objects.dir/Unwind-EHABI.cpp.o
clang++: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
[3/21] Building C object libunwind/src/CMakeFiles/unwind_shared_objects.dir/Unwind-sjlj.c.o
FAILED: libunwind/src/CMakeFiles/unwind_shared_objects.dir/Unwind-sjlj.c.o 
/nix/store/19sfx69d3inys39h88m2b045sasln0dv-clang-wrapper-19.0.0-unstable-2024-05-14/bin/clang -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBUNWIND_LINK_DL_LIB -D_LIBUNWIND_LINK_PTHREAD_LIB -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/libunwind-src-19.0.0-unstable-2024-05-14/libunwind/include -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wno-comment -fdiagnostics-color -O3 -DNDEBUG -fPIC -Werror=return-type -funwind-tables -nostdinc++ -D_DEBUG -UNDEBUG -D_LIBUNWIND_IS_NATIVE_ONLY -Wall -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error -pedantic -fno-rtti -std=c99 -MD -MT libunwind/src/CMakeFiles/unwind_shared_objects.dir/Unwind-sjlj.c.o -MF libunwind/src/CMakeFiles/unwind_shared_objects.dir/Unwind-sjlj.c.o.d -o libunwind/src/CMakeFiles/unwind_shared_objects.dir/Unwind-sjlj.c.o -c /build/libunwind-src-19.0.0-unstable-2024-05-14/libunwind/src/Unwind-sjlj.c
clang: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
In file included from /build/libunwind-src-19.0.0-unstable-2024-05-14/libunwind/src/Unwind-sjlj.c:12:
/build/libunwind-src-19.0.0-unstable-2024-05-14/libunwind/include/unwind.h:19:10: fatal error: 'stddef.h' file not found
   19 | #include <stddef.h>
      |          ^~~~~~~~~~
1 error generated.
```

I'm able to get `libcxx` to compile by adding `llvmPackages_git.clangUseLLVM` as a `nativeBuildInput`, but I'm having issues with overriding `libunwind`.

**Comment by Nanotwerp** [comment]
Could it be possible that the issue's caused by the updated `gnu-install-dirs` patch?

**Comment by RossComputerGuy** [comment]
> Could it be possible that the issue's caused by the updated `gnu-install-dirs` patch?

Unlikely, the update only really updates the patch so that it could apply right. Still functions the same as what we already have.

**Comment by Nanotwerp** [comment]
> > Could it be possible that the issue's caused by the updated `gnu-install-dirs` patch?
> 
> Unlikely, the update only really updates the patch so that it could apply right. Still functions the same as what we already have.

Gotcha! I finally overrode `libunwind`, but the same fix sadly doesn't seem to work for it.

**Comment by EricWF** [comment]
The `stddef.h` header that's missing should be found in the `<clang-install-prefix>/lib/clang/<version>/include/stddef.h`.
On my debian system, the actual path is `/usr/local/lib/clang/19/include/stddef.h`.  

In the first case you're building C++, in the second you're building C. In both cases, Clang should be providing the header.

You can add `-v` to the command invocation and Clang should dump the include paths its using, along with diagnostics about any include paths that were added, but don't exist. For example:

`clang -v -xc /dev/null -fsyntax-only`
```c++
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/13/../../../../x86_64-linux-gnu/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/lib/clang/19/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
```

Could you verify that the lib/clang/<version>/include is present and contains the `stddef.h` header?

**Comment by RossComputerGuy** [comment]
@EricWF This is the result:
```
clang version 19.0.0git
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /nix/store/lkwv21jayla68rijvyncn7ajlddxw4d1-clang-19.0.0-unstable-2024-05-26/bin
Found candidate GCC installation: /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib/gcc/aarch64-unknown-linux-gnu/13.2.0
Found candidate GCC installation: /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib64/gcc/aarch64-unknown-linux-gnu/13.2.0
Selected GCC installation: /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib64/gcc/aarch64-unknown-linux-gnu/13.2.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
clang++: warning: -Wl,-dynamic-linker=/nix/store/10pd749c2v3m35d4hk5xdfylyfr1hzqz-glibc-2.39-5/lib/ld-linux-aarch64.so.1: 'linker' input unused [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: '-L/nix/store/10pd749c2v3m35d4hk5xdfylyfr1hzqz-glibc-2.39-5/lib' [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: '-L/nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib/gcc/aarch64-unknown-linux-gnu/13.2.0' [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: '-L/nix/store/i578g2ws943hl9crhapvy35l6j662xpj-gcc-13.2.0-lib/lib' [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: '-L/nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0//lib' [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: '-L/nix/store/d1yb0arh6k658anwsgl8d5sks6zcqrgm-gcc-13.2.0-libgcc/lib' [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: '-L/nix/store/ivrv22snibgapyj3x4f88zdbqpd99d4x-clang-19.0.0-unstable-2024-05-26-lib/lib' [-Wunused-command-line-argument]
 (in-process)
 "/nix/store/lkwv21jayla68rijvyncn7ajlddxw4d1-clang-19.0.0-unstable-2024-05-26/bin/clang++" -cc1 -triple aarch64-unknown-linux-gnu -fsyntax-only -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name null -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=non-leaf -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu generic -target-feature +v8a -target-feature +fp-armv8 -target-feature +neon -target-abi aapcs -debugger-tuning=gdb -fdebug-compilation-dir=/home/thecomputerguy -v -fcoverage-compilation-dir=/home/thecomputerguy -nostdsysteminc -resource-dir /nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root -idirafter /nix/store/dyyak1vqjbj3q9jihmai954jgzdkarsy-glibc-2.39-5-dev/include -isystem /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0 -isystem /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0/aarch64-unknown-linux-gnu -internal-isystem /nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root/include -ferror-limit 19 -fno-signed-char -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -target-feature +outline-atomics -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -x c /dev/null
clang -cc1 version 19.0.0git based upon LLVM 19.0.0git default target aarch64-unknown-linux-gnu
ignoring nonexistent directory "/nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root/include"
#include "..." search starts here:
#include <...> search starts here:
 /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0
 /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0/aarch64-unknown-linux-gnu
 /nix/store/dyyak1vqjbj3q9jihmai954jgzdkarsy-glibc-2.39-5-dev/include
End of search list.
```

**Comment by Nanotwerp** [comment]
OK, maybe I didn't successfully override `libunwind` after all. After testing with `nix repl`, I found out that adding `pkgs.llvmPackages_git.clangUseLLVM` (meaning version `19.0.0-unstable-2024-04-07`) to its `nativeBuildInputs` allows to `libunwind` to successfully compile. Note that it also fails to compile with the same error if I instead add the `clangUseLLVM` of `19.0.0-unstable-2024-05-26` to `nativeBuildInputs`. This leads me to think the updated `clang` is the actual issue.

`nix-repl> :b inputs.llvmPackages_git.outputs.legacyPackages.x86_64-linux.llvmPackages_git.libunwind`:
```
error: builder for '/nix/store/fzs4mna736pixfk7vm9pb6npfz4bq7j3-libunwind-19.0.0-unstable-2024-05-26.drv' failed with exit code 1;
       last 10 log lines:
       > [18/21] Building CXX object libunwind/src/CMakeFiles/unwind_shared_objects.dir/libunwind.cpp.o
       > FAILED: libunwind/src/CMakeFiles/unwind_shared_objects.dir/libunwind.cpp.o
       > /nix/store/2qiyg3ssivgjdah09ygx394jaiv5acqn-clang-wrapper-19.0.0-unstable-2024-05-26/bin/clang++ -D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER -D_LIBUNWIND_LINK_DL_LIB -D_LIBUNWIND_LINK_PTHREAD_LIB -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/libunwind-src-19.0.0-unstable-2024-05-26/libunwind/include -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -std=c++17 -fPIC -Werror=return-type -funwind-tables -nostdinc++ -D_DEBUG -UNDEBUG -D_LIBUNWIND_IS_NATIVE_ONLY -Wall -Wextra -Wnewline-eof -Wshadow -Wwrite-strings -Wno-unused-parameter -Wno-long-long -Werror=return-type -Wextra-semi -Wundef -Wunused-template -Wformat-nonliteral -Wno-user-defined-literals -Wno-covered-switch-default -Wno-suggest-override -Wno-error -pedantic -fno-rtti  -fstrict-aliasing -fno-exceptions -fno-rtti -MD -MT libunwind/src/CMakeFiles/unwind_shared_objects.dir/libunwind.cpp.o -MF libunwind/src/CMakeFiles/unwind_shared_objects.dir/libunwind.cpp.o.d -o libunwind/src/CMakeFiles/unwind_shared_objects.dir/libunwind.cpp.o -c /build/libunwind-src-19.0.0-unstable-2024-05-26/libunwind/src/libunwind.cpp
       > clang++: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
       > In file included from /build/libunwind-src-19.0.0-unstable-2024-05-26/libunwind/src/libunwind.cpp:12:
       > /build/libunwind-src-19.0.0-unstable-2024-05-26/libunwind/include/libunwind.h:19:10: fatal error: 'stddef.h' file not found
       >    19 | #include <stddef.h>
       >       |          ^~~~~~~~~~
       > 1 error generated.
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/fzs4mna736pixfk7vm9pb6npfz4bq7j3-libunwind-19.0.0-unstable-2024-05-26.drv'.
```

`nix-repl> :b inputs.llvmPackages_git.outputs.legacyPackages.x86_64-linux.llvmPackages_git.libunwind.overri
deAttrs (oldAttrs: { nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [pkgs.llvmPackages_git.clangUseLLVM]; })`:
```
This derivation produced the following outputs:
  dev -> /nix/store/y3hqf9r16b7p7f8xi378wlnm55vx1pag-libunwind-19.0.0-unstable-2024-05-26-dev
  out -> /nix/store/m8mi75ncl7li31ycl1wh72wbfbfpwq5f-libunwind-19.0.0-unstable-2024-05-26
```

**Comment by EricWF** [comment]
I think this bit of output is the key:

> ignoring nonexistent directory "/nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root/include"

My guess is that's the directory where `stddef.h` is meant to live.

**Comment by Nanotwerp** [comment]
> @EricWF This is the result:
> 
> ```
> clang version 19.0.0git
> Target: aarch64-unknown-linux-gnu
> Thread model: posix
> InstalledDir: /nix/store/lkwv21jayla68rijvyncn7ajlddxw4d1-clang-19.0.0-unstable-2024-05-26/bin
> Found candidate GCC installation: /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib/gcc/aarch64-unknown-linux-gnu/13.2.0
> Found candidate GCC installation: /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib64/gcc/aarch64-unknown-linux-gnu/13.2.0
> Selected GCC installation: /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib64/gcc/aarch64-unknown-linux-gnu/13.2.0
> Candidate multilib: .;@m64
> Selected multilib: .;@m64
> clang++: warning: -Wl,-dynamic-linker=/nix/store/10pd749c2v3m35d4hk5xdfylyfr1hzqz-glibc-2.39-5/lib/ld-linux-aarch64.so.1: 'linker' input unused [-Wunused-command-line-argument]
> clang++: warning: argument unused during compilation: '-L/nix/store/10pd749c2v3m35d4hk5xdfylyfr1hzqz-glibc-2.39-5/lib' [-Wunused-command-line-argument]
> clang++: warning: argument unused during compilation: '-L/nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/lib/gcc/aarch64-unknown-linux-gnu/13.2.0' [-Wunused-command-line-argument]
> clang++: warning: argument unused during compilation: '-L/nix/store/i578g2ws943hl9crhapvy35l6j662xpj-gcc-13.2.0-lib/lib' [-Wunused-command-line-argument]
> clang++: warning: argument unused during compilation: '-L/nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0//lib' [-Wunused-command-line-argument]
> clang++: warning: argument unused during compilation: '-L/nix/store/d1yb0arh6k658anwsgl8d5sks6zcqrgm-gcc-13.2.0-libgcc/lib' [-Wunused-command-line-argument]
> clang++: warning: argument unused during compilation: '-L/nix/store/ivrv22snibgapyj3x4f88zdbqpd99d4x-clang-19.0.0-unstable-2024-05-26-lib/lib' [-Wunused-command-line-argument]
>  (in-process)
>  "/nix/store/lkwv21jayla68rijvyncn7ajlddxw4d1-clang-19.0.0-unstable-2024-05-26/bin/clang++" -cc1 -triple aarch64-unknown-linux-gnu -fsyntax-only -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name null -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=non-leaf -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu generic -target-feature +v8a -target-feature +fp-armv8 -target-feature +neon -target-abi aapcs -debugger-tuning=gdb -fdebug-compilation-dir=/home/thecomputerguy -v -fcoverage-compilation-dir=/home/thecomputerguy -nostdsysteminc -resource-dir /nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root -idirafter /nix/store/dyyak1vqjbj3q9jihmai954jgzdkarsy-glibc-2.39-5-dev/include -isystem /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0 -isystem /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0/aarch64-unknown-linux-gnu -internal-isystem /nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root/include -ferror-limit 19 -fno-signed-char -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -target-feature +outline-atomics -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -x c /dev/null
> clang -cc1 version 19.0.0git based upon LLVM 19.0.0git default target aarch64-unknown-linux-gnu
> ignoring nonexistent directory "/nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root/include"
> #include "..." search starts here:
> #include <...> search starts here:
>  /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0
>  /nix/store/8y16aj4mk0rbck2hlg538322d5r1gi7m-gcc-13.2.0/include/c++/13.2.0/aarch64-unknown-linux-gnu
>  /nix/store/dyyak1vqjbj3q9jihmai954jgzdkarsy-glibc-2.39-5-dev/include
> End of search list.
> ```

This is in contrast to the result that the `clang` of the current `llvmPackages_git` gives, which is:

```
clang version 19.0.0git
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /nix/store/lbpf9gjfnch6cmdxg14w95167fsg6h3s-clang-19.0.0-unstable-2024-04-07/bin
Found candidate GCC installation: /nix/store/14c6s4xzhy14i2b05s00rjns2j93gzz4-gcc-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0
Found candidate GCC installation: /nix/store/14c6s4xzhy14i2b05s00rjns2j93gzz4-gcc-13.2.0/lib64/gcc/x86_64-unknown-linux-gnu/13.2.0
Selected GCC installation: /nix/store/14c6s4xzhy14i2b05s00rjns2j93gzz4-gcc-13.2.0/lib64/gcc/x86_64-unknown-linux-gnu/13.2.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
 (in-process)
 "/nix/store/lbpf9gjfnch6cmdxg14w95167fsg6h3s-clang-19.0.0-unstable-2024-04-07/bin/clang-19" -cc1 -triple x86_64-unknown-linux-gnu -fsyntax-only -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name null -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=/home/lokottara/Code/nixpkgs -v -fcoverage-compilation-dir=/home/lokottara/Code/nixpkgs -nostdsysteminc -resource-dir /nix/store/hpggqnfkyc0mad28jqrwhqm8vzfdf1i4-clang-wrapper-19.0.0-unstable-2024-04-07/resource-root -idirafter /nix/store/fwh4fxd747m0py3ib3s5abamia9nrf90-glibc-2.39-52-dev/include -internal-isystem /nix/store/hpggqnfkyc0mad28jqrwhqm8vzfdf1i4-clang-wrapper-19.0.0-unstable-2024-04-07/resource-root/include -Wno-unused-command-line-argument -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcolor-diagnostics -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -x c /dev/null
clang -cc1 version 19.0.0git based upon LLVM 19.0.0git default target x86_64-unknown-linux-gnu
#include "..." search starts here:
#include <...> search starts here:
 /nix/store/hpggqnfkyc0mad28jqrwhqm8vzfdf1i4-clang-wrapper-19.0.0-unstable-2024-04-07/resource-root/include
 /nix/store/fwh4fxd747m0py3ib3s5abamia9nrf90-glibc-2.39-52-dev/include
End of search list.
```

**Comment by RossComputerGuy** [comment]
@EricWF Yes, that appears so. It is a symlink:
```
lrwxrwxrwx 1 root root 101 Jan  1  1970 /nix/store/r0xch7m9d9kx40b501xbqhxnycrh3w62-clang-wrapper-19.0.0-unstable-2024-05-26/resource-root/include -> /nix/store/ivrv22snibgapyj3x4f88zdbqpd99d4x-clang-19.0.0-unstable-2024-05-26-lib/lib/clang/19/include
```

But the directory the symlink points to doesn't exist but it is the directory you were referring to. All of our headers are actually in `/nix/store/lkwv21jayla68rijvyncn7ajlddxw4d1-clang-19.0.0-unstable-2024-05-26/lib/clang/19/include/`.

**Comment by RossComputerGuy** [comment]
I pinpoined the issue being a part of when I [commonified LLVM](https://github.com/NixOS/nixpkgs/commit/dc4d5c21937ec81d41e0211d74d6e562955bf5be). I fixed the issue.

**Comment by EricWF** [comment]
> I pinpoined the issue being a part of when I [commonified LLVM](https://github.com/NixOS/nixpkgs/commit/dc4d5c21937ec81d41e0211d74d6e562955bf5be#diff-ba04eed4b6799d1eef9fb33b5210a045f4cb584f51399b9d1d1899bc8d2fcc82R88-R92). I fixed the issue.

Fantastic!

**Comment by Nanotwerp** [comment]
> I pinpoined the issue being a part of when I [commonified LLVM](https://github.com/NixOS/nixpkgs/commit/dc4d5c21937ec81d41e0211d74d6e562955bf5be#diff-ba04eed4b6799d1eef9fb33b5210a045f4cb584f51399b9d1d1899bc8d2fcc82R88-R92). I fixed the issue.

Nice! Just updated to [6573d](https://github.com/NixOS/nixpkgs/commit/6573d708078d1deac209591da825252a83038f04) and the issue is fixed!

**Comment by RossComputerGuy** [comment]
@ofborg build llvmPackages_git.clang

**Comment by RossComputerGuy** [comment]
@ofborg build llvmPackages_git.clang

**Comment by RossComputerGuy** [comment]
@ofborg eval

**Comment by RossComputerGuy** [comment]
Build is good after updating, ready to merge.

**Review by tomberek — APPROVED** [review]
Built on x86_64-linux

**Pull request merged**